### PR TITLE
[#335] As a developer, I can upload the build to Firebase App Distribution with Fastlane Swift

### LINKED ISSUE #335: As a developer, I can upload the build to Firebase App Distribution with Fastlane Swift
## Why

Currently, iOS developers need to learn at least basic Ruby to understand or write/update Fastlane configurations. To reduce the effort, we will create Fastlane in Swift language. So everyone in the iOS Chapter can use easier.

## Acceptance Criteria

- Use Fastlane's [firebase_app_distribution](https://github.com/fastlane/fastlane-plugin-firebase_app_distribution) plugin to upload the build to Firebase App Distribution.

### PR DESCRIPTION
**Pull request by vnntsu**
resolves #335

## What happened

- Add Firebase distribution `plugin` to our project.
- Create `Distribution` to be in charge of uploading build to Firebase.
 
## Insight

- Since we're going to reuse `Environment` and `BuildType` (inside `Build`) at `Distribution`, so I move it to `Constant` instead.
- Insert firebase CLI token to `Secret`
- Create a function to upload based on the environment as a parameter.
- Create 2 lanes for build and upload for `Staging` and `Production`
 
## Proof Of Work

Screen Shot 2022-09-26 at 15 52 35

Screen Shot 2022-09-26 at 16 09 59

### CODE REVIEW

File: fastlane/Constants/Constant.swift [hunk 1]
```diff
@@ -17,12 +17,66 @@ enum Constant {
...
+        case production = ""
+
+        var productName: String { "\(Constant.productName) \(rawValue)".trimmed }
+
+        var scheme: String { "\(Constant.scheme) \(rawValue)".trimmed }
+
+        var bundleId: String {
+            let bundleId = Constant.bundleId
+            switch self {
+            case .staging: return "\(Constant.bundleId).\(rawValue.lowercased())"
```
Comment (suho) [inline fastlane/Constants/Constant.swift:50]: I think we might need to create a bundle id constant for staging as well, not every time we have
- product: co.nimblehq.app
- staging: co.nimblehq.app.staging 

We might have something like `co.nimblehq.app-staging` for example.

What do you think?

File: fastlane/swift/Actions.swift [hunk 2]
```diff
@@ -1,16 +1 @@
...
-//
-//  ** NOTE **
-//  This file is provided by fastlane and WILL be overwritten in future updates
-//  If you want to add extra functionality to this project, create a new file in a
-//  new group so that it won't be marked for upgrade
-//
-
 import Foundation
-
-// Please don't remove the lines below
```
Comment (suho) [inline fastlane/swift/Actions.swift:14]: I think we don't need to change this file

Comment (vnntsu) [reply fastlane/swift/Actions.swift:14]: it's changed by the `Fastlane` automatically after adding the plugin. These changes are all `remove`, so I think it's fine

**Review by ducbm051291 — APPROVED** [review]
Agree with @suho 's suggestion. The rest LGTM

File: fastlane/Constants/Constant.swift (hunk 1 above)
Comment (vnntsu) [reply fastlane/Constants/Constant.swift]: @suho Yes, I also think it's better with explicit define. Updated in [1c5b9d7](https://github.com/nimblehq/ios-templates/pull/354/commits/1c5b9d7fcc401f4a9a004872b69ead50a7bd6e4c)

**Review by suho — APPROVED** [review]

**Review by nkhanh44 — APPROVED** [review]

**Pull request merged**